beaker/src/stamp_manager.rs

```diff
@@ -0,0 +1,339 @@
+//! Stamp file management for Make-compatible incremental builds
+//!
+//! This module handles the creation and management of stamp files that enable
+//! Make to perform accurate incremental builds. Each stamp file contains a
+//! deterministic hash based only on inputs that affect the byte-level output
+//! of that stage.
+
+use anyhow::{anyhow, Result};
+use sha2::{Digest, Sha256};
+use std::fs;
+use std::path::{Path, PathBuf};
+
+use crate::cache_common::{calculate_md5, get_cache_base_dir};
+use crate::config::{CutoutConfig, DetectionConfig};
+
+/// Generate a deterministic configuration hash for detection
```

Comment:
Is it possible we could somehow annotate which fields are relevant in the definition of the config itself, and then have this autogenerated? So the source of truth stays at the definition site. Is there a crate for that? Or can we write something?